osf/management/commands/rebrand_preprint_provider.py

```diff
@@ -0,0 +1,108 @@
+
+from scripts import utils as script_utils
+from osf.management.commands.reindex_preprint_provider import reindex_provider
+from osf.models import PreprintProvider
+from osf.utils.migrations import disable_auto_now_fields
+
+logger = logging.getLogger(__name__)
+
+def rebrand_provider(src_id, dst_id):
+    assert src_id != 'osf', 'Cannot rebrand OSF Preprints'
+    src_prov = PreprintProvider.load(src_id)
+    dst_prov = PreprintProvider.load(dst_id)
+    assert src_prov, 'Unable to find provider {}'.format(src_id)
+    assert dst_prov, 'Unable to find provider {}'.format(dst_id)
+    assert set(src_prov.subjects.values_list('text', flat=True)) == set(dst_prov.subjects.values_list('text', flat=True)), 'Provider subjects do not match'
+    assert dst_prov.asset_files.filter(name='square_color_no_transparent').exists(), 'Invalid assets on {}'.format(dst_id)
+    assert dst_prov.access_token, 'Destination Provider must have a SHARE access token'
+
+    logger.info('Updating {}\'s preprints to provider {}'.format(src_id, dst_id))
+    src_prov.preprints.update(provider_id=dst_prov.id)
+
+    logger.info('Updating preprint subjects with {} equivalent subjects'.format(dst_id))
+    target_preprints = dst_prov.preprints.all()
+    pbar = progressbar.ProgressBar(maxval=target_preprints.count() or 1).start()
+    for i, pp in enumerate(target_preprints, 1):
+        pbar.update(i)
+        # M2M .set does not require .save
+        pp.subjects.set(dst_prov.subjects.filter(text__in=list(pp.subjects.values_list('text', flat=True))))
+    pbar.finish()
+
+    logger.info('Updating {} moderators'.format(dst_id))
+    dst_prov.get_group('admin').user_set.set(src_prov.get_group('admin').user_set.all())
+    dst_prov.get_group('moderator').user_set.set(src_prov.get_group('moderator').user_set.all())
+
+def delete_old_provider(src_id):
+    src_prov = PreprintProvider.load(src_id)
+    assert src_prov.preprints.count() == 0, 'Provider {} still has preprints'.format(src_id)
+    assert src_prov.subjects.annotate(pc=Count('preprints')).filter(pc__gt=0).count() == 0, 'Provider {} still has used subjects'.format(src_id)
+
+    # I don't trust CASCADE deletes to be set up correctly
+    logger.warn('Deleting Assets: {}'.format(src_prov.asset_files.annotate(pc=Count('providers')).filter(pc=1).delete()))
+    logger.warn('Deleting Groups: {}'.format(src_prov.group_objects.delete()))
+    logger.warn('Deleting Subjects: {}'.format(src_prov.subjects.all().delete()))
+    logger.warn('Deleting Subscriptions: {}'.format(src_prov.notification_subscriptions.all().delete()))
+    logger.warn('Deleting Digests: {}'.format(src_prov.notificationdigest_set.all().delete()))
+    logger.warn('Deleting Provider: {}'.format(src_prov.delete()))
+
+def reindex_share(dst_id, dry_run):
+    dst_prov = PreprintProvider.load(dst_id)
+    if dry_run:
+        logger.info('Would send {} preprints to SHARE...'.format(dst_prov.preprints.count()))
+    else:
+        reindex_provider(dst_prov)
```

Comment:
We don't send any provider-specific info to elastic search, right? Maybe the url for the preprint, but I see you're going to set that up to redirect.

Reply:
[Nope](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/search/elastic_search.py#L426-L459).